src/Simplex/Messaging/Agent.hs

```diff
@@ -1267,6 +1277,7 @@ runSmpQueueMsgDelivery c@AgentClient {subQ} ConnData {connId} sq (Worker {doWork
         let mId = unId msgId
             ri' = maybe id updateRetryInterval2 msgRetryState ri
         withRetryLock2 ri' qLock $ \riState loop -> do
+          lift $ waitForUserNetwork c
```

Comment:
why not before `beginAgentOperation c AOSndNetwork`, or at least after `endAgentOperation c AOMsgDelivery` before withRetryLock2? Not saying it's not correct, just wondering why it's not earlier